pkg/syncer/fullsync.go

```diff
@@ -284,13 +288,94 @@ func CsiFullSync(ctx context.Context, metadataSyncer *metadataSyncInformer, vc s
 	log.Infof("FullSync for VC %s: end", vc)
 	return nil
 }
 
+// cnsVolumeOperationRequestCRFullSync patches CNSVolumeOperationsRequest CRs for in-progress PVC creation
+func cnsVolumeOperationRequestCRFullSync(ctx context.Context, metadataSyncer *metadataSyncInformer, vc string) {
+	log := logger.GetLogger(ctx)
+	log.Debugf("Starting CnsVolumeOperationRequest CR full sync.")
+	namespace := "vmware-system-csi"
+	cnsVolumeOperationRequestList := &cnsvolumeoperationrequestv1alpha1.CnsVolumeOperationRequestList{}
+	err := metadataSyncer.cnsOperatorClient.List(ctx, cnsVolumeOperationRequestList, client.InNamespace(namespace))
+	if err != nil {
+		log.Errorf("failed to list %s CR from supervisor namespace %q. Error: %+v",
+			cnsvolumeoperationrequestv1alpha1.CRDSingular, namespace, err)
+		return
+	}
+	cnsVolumeOpRequestCRsToSvcNamespace := make(map[string][]string)
+	for _, cr := range cnsVolumeOperationRequestList.Items {
+		svcNamespace := cr.Status.StorageQuotaDetails.Namespace
+		if _, ok := cnsVolumeOpRequestCRsToSvcNamespace[cr.Status.StorageQuotaDetails.Namespace]; ok {
+			cnsVolumeOpRequestCRsToSvcNamespace[svcNamespace] = append(cnsVolumeOpRequestCRsToSvcNamespace[svcNamespace], cr.Name)
+		} else {
+			cnsVolumeOpRequestCRsToSvcNamespace[svcNamespace] = []string{cr.Name}
+		}
+	}
+	for ns, cnsVolumeOpRequestCRList := range cnsVolumeOpRequestCRsToSvcNamespace {
+		storagePolicyUsageList := &storagepolicyusagev1alpha1.StoragePolicyUsageList{}
+		err = metadataSyncer.cnsOperatorClient.List(ctx, storagePolicyUsageList, client.InNamespace(ns))
+		if err != nil {
+			log.Errorf("failed to list %s CR from supervisor namespace %q. Error: %+v",
+				storagepolicyusagev1alpha1.CRDSingular, ns, err)
+			return
+		}
+		storagePolicyUsageCR := &storagepolicyusagev1alpha1.StoragePolicyUsage{}
+		if len(storagePolicyUsageList.Items) > 0 {
+			for _, item := range storagePolicyUsageList.Items {
+				policyId := item.Spec.StoragePolicyId
+				scName := item.Spec.StorageClassName
+				if item.Spec.StoragePolicyId == policyId && item.Spec.StorageClassName == scName {
+					log.Debugf("Found storagePolicyUsage CR with matching "+
+						"storagePolicyId: %q & storageClassName: %q in namespace: %q", policyId, scName,
+						ns)
+					oldStoragePolicyUsageCR := item.DeepCopy()
+					newStoragePolicyUsageCR := item.DeepCopy()
+					used := &resource.Quantity{}
+					// Get K8s PVs in State "Bound".
+					k8sPVsInBoundState, err := getBoundPVs(ctx, metadataSyncer)
+					if err != nil {
+						log.Errorf("Failed to get PVs from kubernetes. Err: %+v", err)
+						return
+					}
+					for _, pv := range k8sPVsInBoundState {
+						used.Add(*resource.NewQuantity(int64(pv.Size()),
+							oldStoragePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage.Used.Format))
+					}
+					storagePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage.Used = used
+					err = patchStoragePolicyUsage(ctx, metadataSyncer.cnsOperatorClient, oldStoragePolicyUsageCR,
```

Comment:
we are directly patching storagepolicyusage CRs here, whereas the description says that this function "patches CNSVolumeOperationsRequest CRs "

Reply:
Updated